rotkehlchen/data/import_utils.py

```diff
@@ -0,0 +1,341 @@
+                cls.db.msg_aggregator.add_warning(
+                    'Skipped some rows because couldn\'t find amounts or it was zero',
+                )
+                return []
+
+        # Grouping identical trades
+        grouped_trades_dicts = []
+        for _, grouped_trades in groupby(sorted(prepared_trade_dicts, key=cls.unique_trades_key), key=cls.unique_trades_key):  # noqa: E501
+            grouped_trades_dicts.append(list(grouped_trades))
+
+        # Combining identical trades
+        unique_trade_dicts = []
+        for trades_dict_group in grouped_trades_dicts:
+            result_trade_dict = trades_dict_group[0]
+            for trade_dict in trades_dict_group[1:]:
+                result_trade_dict['to_amount'] += trade_dict['to_amount']
+                result_trade_dict['from_amount'] += trade_dict['from_amount']
+                if result_trade_dict['fee_amount'] is not None:
+                    result_trade_dict['fee_amount'] += trade_dict['fee_amount']
+            unique_trade_dicts.append(result_trade_dict)
+
+        # Creating Trade classes.
+        trades = []
+        for trade_dict in unique_trade_dicts:
+            rate = FVal(trade_dict['to_amount']) / FVal(trade_dict['from_amount'])
+            trades.append(Trade(
+                timestamp=timestamp,
+                location=Location.BINANCE,
+                base_asset=trade_dict['to_asset'],
+                quote_asset=trade_dict['from_asset'],
+                trade_type=trade_dict['trade_type'],
+                amount=trade_dict['to_amount'],
+                rate=Price(rate),
+                fee=trade_dict['fee_amount'],
+                fee_currency=trade_dict['fee_asset'],
+                link='',
+            ))
+        return trades
+
+    @classmethod
+    def process_entries(cls, timestamp: Timestamp, data: List[BinanceCsvRow]) -> int:
+        trades = cls.process_trades(timestamp=timestamp, data=data)
+        cls.db.add_trades(trades)
+        return len(trades)
+
+
+class BinanceDepositWithdrawEntry(BinanceSingleEntry):
+    """This class processes Deposit and Withdraw actions
+        which are AssetMovements in the internal representation"""
+
+    available_operations = ['Deposit', 'Withdraw']
+
+    @classmethod
+    def process_entry(cls, timestamp: Timestamp, data: BinanceCsvRow) -> None:
+        amount = data['Change']
+        asset = data['Coin']
+        category = AssetMovementCategory.DEPOSIT if data['Operation'] == 'Deposit' else AssetMovementCategory.WITHDRAWAL  # noqa: E501
+        if category == AssetMovementCategory.WITHDRAWAL:
+            amount = -amount
+
+        asset_movement = AssetMovement(
+            location=Location.BINANCE,
+            category=category,
+            address=None,
+            transaction_id=None,
+            timestamp=timestamp,
+            asset=asset,
+            amount=AssetAmount(amount),
+            fee=Fee(ZERO),
+            fee_asset=A_USD,
+            link='',
```

Comment:
Oh hm ... use the `link` one then. Kind of sucks and we should probably change this in the future for consistency but won't add new field to AssetMovement just for this.

So between adding nothing, and using a special note for `link` instead of just `''` I think using a special note makes more sense.

Reply:
Alright. I was afraid that link has another purpose that is not suitable for notes. But if it's okay, then surely I will add the note here